hazelcast/src/main/java/com/hazelcast/internal/partition/operation/MigrationRequestOperation.java

```diff
@@ -75,67 +76,15 @@ public MigrationRequestOperation(MigrationInfo migrationInfo, int partitionState
     }
 
     @Override
-    public void run() {
-        verifyMasterOnMigrationSource();
-        NodeEngineImpl nodeEngine = (NodeEngineImpl) getNodeEngine();
-
-        Address source = migrationInfo.getSource();
-        Address destination = migrationInfo.getDestination();
-        verifyExistingTarget(nodeEngine, destination);
-
-        if (destination.equals(source)) {
-            getLogger().warning("Source and destination addresses are the same! => " + toString());
-            setFailed();
-            return;
-        }
-
-        InternalPartition partition = getPartition();
-        verifySource(nodeEngine.getThisAddress(), partition);
-
-        setActiveMigration();
-
-        if (!migrationInfo.startProcessing()) {
-            getLogger().warning("Migration is cancelled -> " + migrationInfo);
-            setFailed();
-            return;
-        }
-
-        try {
-            executeBeforeMigrations();
-            namespacesContext = new ServiceNamespacesContext(nodeEngine, getPartitionReplicationEvent());
-            ReplicaFragmentMigrationState migrationState = fragmentedMigrationEnabled
-                    ? createNextReplicaFragmentMigrationState()
-                    : createAllReplicaFragmentsMigrationState();
-            invokeMigrationOperation(destination, migrationState, true);
-            returnResponse = false;
-        } catch (Throwable e) {
-            logThrowable(e);
-            setFailed();
-        } finally {
-            migrationInfo.doneProcessing();
-        }
-    }
-
-    @Override
-    void onMigrationStart() {
-        ((OperationServiceImpl) getNodeEngine().getOperationService()).onStartAsyncOperation(this);
-        super.onMigrationStart();
-    }
-
-    @Override
-    void onMigrationComplete(boolean result) {
-        try {
-            super.onMigrationComplete(result);
-        } finally {
-            ((OperationServiceImpl) getNodeEngine().getOperationService()).onCompletionAsyncOperation(this);
-        }
+    public CallStatus call() {
+        return new OffloadImpl();
```

Comment:
I'm not sure this is going to work since now the replication data is collected on the offloaded thread and not on the partition thread. cc @mdogan

Reply:
The offload.start is still called on the partition thread.

Reply:
False alarm.